model/src/main/java/com/expediagroup/streamplatform/streamregistry/model/Principal.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (C) 2018-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class Principal {
+  private String name;
+}
```

Comment:
Does it need to be mutable? Also non-null `name`?

```suggestion
import lombok.Value;

@Value
public class Principal {
  String name;
}
```

Reply:
OK - taking this change for both Role and Principal.